modules/dnn/src/opencl/mvn.cl

```diff
@@ -150,6 +153,58 @@ __kernel void MVN(__global const Dtype* src,
     store(dst_vec, dst, index);
 }
 
+#elif defined KERNEL_MVN_GROUP
+
+__kernel void MVN_GROUP(__global const Dtype* src,
+                            const int rows,
+                            const int cols,
+                            const Dtype eps,
+                            __global const Dtype* mean,
+                            __global const Dtype* dev,
+                            __global const Dtype* bnorm_weight,
+                            __global const Dtype* bnorm_bias,
+                            const int channels,
+                            const int num_groups,
+                            const float relu_slope,
+                            __global Dtype* dst)
+{
+    int x = get_global_id(0);
+    int y = get_global_id(1) * NUM;
+    int index = x * cols + y;
+
+    if (x >= rows || y >= cols)
+        return;
+
+    int group_size = channels / num_groups;
+    int step = norm_size / group_size;
+    int channel_index = x % num_groups * group_size + y / step
+    Dtype mean_val = mean[x];
+    Dtype dev_val = dev[x];
+    Dtype alpha;
+#ifdef NORM_VARIANCE
+    alpha = 1 / sqrt(eps + dev_val);
+#else
+    alpha = 1;
+#endif
+
+    Dtype w = 1.f, b = 0.f;
+#ifdef FUSE_GROUP_NORM
```

Comment:
Seems like you wanted to fuse a batch norm, not group norm?

Reply:
Actually, I created the MVN_GROUP function to accommodate group normalization. This is because, according to the latest ONNX definition, the shapes of mean and variance are different from the shapes of scale and bias. Hence, group normalization requires two integer parameters (channel and num_groups) to execute the MVN calculation. The original MVN function probably isn't up to this task. I wrote #ifdef FUSE_GROUP_NORM in reference to the previous MVN function, but if this new function is only used by group norm, I think I can remove the macro definition. Batch norm can just use the previous MVN function.

Reply:
This is a dedicated function that comes with scale and bias, so we don't need `FUSE_BNORM` nor `FUSE_GROUP_NORM`. In this case, names of parameters can be changed to `weight` or `scale` without the `bnorm` prefix.